Dockerfile

```diff
@@ -1,47 +1,43 @@
-FROM ubuntu:16.04
-LABEL maintainer="johan@linkdata.se"
-
-ENV DOTNET_CLI_TELEMETRY_OPTOUT 1
-ENV NUGET_XMLDOC_MODE skip
-RUN export DEBIAN_FRONTEND='noninteractive' && \
-	apt-get update -q && \
-	apt-get install -qy \
-		apt-utils \
-		apt-transport-https \
-		libboost-all-dev \
-		swi-prolog-nox \
-		libaio1 \
-		libstdc++6 \
-		curl \
-		unzip \
-        && \
-	apt-get install -qy software-properties-common && \
-	add-apt-repository ppa:ubuntu-toolchain-r/test && \
-	apt-get update -q && \
-	apt-get install gcc-4.9 -qy && \
-	apt-get upgrade libstdc++6 -qy && \		
-	echo "deb [arch=amd64] https://apt-mo.trafficmanager.net/repos/dotnet-release/ xenial main" > /etc/apt/sources.list.d/dotnetdev.list && \
-	apt-key adv --keyserver apt-mo.trafficmanager.net --recv-keys 417A0893 && \
-	apt-get update -q && \
-	apt-get install -qy dotnet-dev-1.0.4 dotnet-sdk-2.0.0 && \
-	mkdir /Starcounter.Nova.Samples && \
-	mkdir dotnet-warmup && \
-	cd dotnet-warmup && \
-	dotnet new xunit && \
-	cd .. && \
-	rm -rf dotnet-warmup && \
-	apt-get clean && \
-
-RUN curl -L -o binaries.zip https://www.myget.org/F/starcounter/api/v2/package/runtime.linux-x64.runtime.native.Starcounter.Bluestar2/2.0.2 && \
-	unzip binaries.zip
-
-ENV PATH $PATH:/runtimes/linux-x64/native
-ENV LD_LIBRARY_PATH /runtimes/linux-x64/native
-
-COPY Program.cs /Starcounter.Nova.Samples
-COPY Starcounter.Nova.Samples.csproj /Starcounter.Nova.Samples
-COPY NuGet.Config /Starcounter.Nova.Samples
-
-CMD cd /Starcounter.Nova.Samples && \
-	dotnet restore && \
-	dotnet run
+FROM microsoft/dotnet:2.1-sdk
+
+ENV BLUESTAR_PACKAGE https://www.myget.org/F/starcounter/api/v2/package/runtime.linux-x64.runtime.native.Starcounter.Bluestar2/2.0.2
+
+WORKDIR /app
+
+# Copy csproj, NuGet.Config, and restore as distinct layers
+COPY *.csproj ./
+COPY NuGet.Config ./
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+RUN dotnet publish -c Release -o out
+
+# Install Bluestar binaries
+WORKDIR /opt/starcounter/bluestar
+RUN apt-get update && apt-get install -y \
+    && --no-install-recommends apt-utils \
+    && unzip \ 
+    && swi-prolog-nox \
+    && libaio1
+
+# Can this be simplified into one line? Should be possible to pipe
```

Comment:
I wasn't able to make `nuget install` work properly because it requires a later version of `nuget` than the one available for Linux. The later version has to be invoked with mono. When I try, I get this:

```
Unhandled Exception:
System.TypeLoadException: Could not load type 'System.AppContext' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeLoadException: Could not load type 'System.AppContext' from assembly 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
```

I could spend more time on fixing this, but seeing as installing manually like this is temporary, I don't think it's worth it.